### ISSUE TITLE
gccgo: can shift by string

### ISSUE THREAD

**Issue by dvyukov** [golang/go#12936]
gccgo successfully compiles the following program:
```go
package e
const c = 0>>""
```

while gc says:
```
/tmp/test.go:2: cannot convert "" to type uint
/tmp/test.go:2: invalid operation: 0 >> "" (shift count type string, must be unsigned integer)
```

gc seems to be right.
gcc on rev 228818, gofrontend on a4bcd319d98ddc52b3e7d16ec87d92aad868ab05

**Comment by dvyukov** [comment]
@ianlancetaylor

**Comment by paranoiacblack** [comment]
Fixed by w/e fixed #12935.
